### ISSUE TITLE
Two different "send link(s)" dialogs

### ISSUE THREAD

**Issue by skynick11** [Zimbra-Community/owncloud-zimlet#226]
There are two "send link(s)" dialogs in two different situations those implement the same functionality, use the same information, but their implementations are not unified.

One is the "send link(s)" option in the file list in the zimlet's tab, whose creates a link on a nextcloud/owncloud side and inserts this link into new mail message:
![link from zimlet tab](https://user-images.githubusercontent.com/37669204/65394151-9ff35980-dd92-11e9-88e7-3f1817fbadae.png)

The second is in the "attach" option of the zimbra's mail Compose dialog:
![link from Mail Compose](https://user-images.githubusercontent.com/37669204/65394150-9ff35980-dd92-11e9-8ddd-eb860fad188b.png)

Could you please
1. Add the date picker dialog to the attach window from the mail Compose dialog (i.e. like it's done in the "send link(s)" window of zimlet's tab).
2. Unify the date format in both windows, for example, as YYYY-MM-DD, since it does not depend on the localization (i.e. as in the attach window from the mail Compose dialog), or automatically determine the format based on the current localization.
3. Autofill the link's expiration date field, so user should clean up this field if he wants to make the link persistent. (It's the default behavior of the Nexcloud UI: it creates the link with the expiration date +7 days to the current date, but during the creation a user can manually modify this field or manually clean up it to make the link persistent).
4. It's also possible to unify labels and to use the "Public link to file/folder" string from the attach window from the mail Compose dialog instead of the simple "Public" one.

**Comment by barrydegraaff** [comment]
yeah it is because the one in the mail composer is created using Zimbra own UI framework (dwt) and the one 
in the application tab is HTML5. 

I will keep this open, but I do not expect to change this soon. 

Kind regards, 

Barry de Graaff 
Zeta Alliance 
Co-founder & Developer 
zetalliance.org | github.com/Zimbra-Community 

Signal: +31 617 220 227 
Fingerprint: 97f4694a1d9aedad012533db725ddd156d36a2d0

**Comment by skynick11** [comment]
> yeah it is because the one in the mail composer is created using Zimbra own UI framework (dwt) and the one in the application tab is HTML5. I will keep this open, but I do not expect to change this soon. 

Could you at least do the autofill of the link's expiration date (on both dialogs)?  (to unify behavior with Nextcloud itself: by default it adds +7 days to the current date)

**Comment by skynick11** [comment]
Points 3 and 4 will be done in #231
